index.d.ts

```diff
@@ -10,12 +10,12 @@ declare namespace WebSocketStream {
   type WebSocketDuplex = Duplex & { socket: WebSocket };
 
   class Server extends WebSocket.Server {
-    on(event: 'connection', cb: (this: WebSocket, socket: WebSocket, request: http.IncomingMessage) => void): this;
-    on(event: 'error', cb: (this: WebSocket, error: Error) => void): this;
-    on(event: 'headers', cb: (this: WebSocket, headers: string[], request: http.IncomingMessage) => void): this;
-    on(event: 'listening', cb: (this: WebSocket) => void): this;
-    on(event: 'stream', cb: (this: WebSocket, stream: WebSocketDuplex, request: http.IncomingMessage) => void): this;
-    on(event: string | symbol, listener: (this: WebSocket, ...args: any[]) => void): this;
+    on(event: 'connection', cb: (this: WebSocket.Server, socket: WebSocket, request: http.IncomingMessage) => void): this;
+    on(event: 'error', cb: (this: WebSocket.Server, error: Error) => void): this;
+    on(event: 'headers', cb: (this: WebSocket.Server, headers: string[], request: http.IncomingMessage) => void): this;
+    on(event: 'listening', cb: (this: WebSocket.Server) => void): this;
+    on(event: 'stream', cb: (this: WebSocket.Server, stream: WebSocketDuplex, request: http.IncomingMessage) => void): this;
+    on(event: string | symbol, listener: (this: WebSocket.Server, ...args: any[]) => void): this;
   }
 
   function createServer(opts?: WebSocket.ServerOptions, callback?: () => void): Server;
```

Comment:
```suggestion
  function createServer(opts?: WebSocket.ServerOptions, callback?: (...args: any[]) => void): Server;
```
I don't mind raising this as a separate PR, but this callback can take parameters,

if you look whats passed into the callback you can see that this is passed into ` this.on('stream', cb)` in server.js

which is a method on its super class  WebSocketServer which is part of the `ws` package

![Screenshot 2024-03-18 at 14 00 45](https://github.com/max-mapper/websocket-stream/assets/102040570/26054535-69e8-4e09-b2ca-7101f7ec7caa)
In there they have many different callback types

Reply:
my brother in christ, this has been open for 4 years now

Reply:
Ah actually there is another ticket raised about this here https://github.com/max-mapper/websocket-stream/issues/158 - don't worry

Reply:
Changes look Good!